securesystemslib/gpg/constants.py

```
@@ -25,30 +26,41 @@
 
 log = logging.getLogger(__name__)
 
-# By default, we assume and test that gpg2 exists. Otherwise, we assume gpg
-# exists.
-GPG_COMMAND = "gpg2"
-GPG_VERSION_COMMAND = GPG_COMMAND + " --version"
-FULLY_SUPPORTED_MIN_VERSION = "2.1.0"
 
-HAVE_GPG = True
-NO_GPG_MSG = "GPG support requires a GPG command, {} version {} or newer is" \
-  " fully supported.".format(GPG_COMMAND, FULLY_SUPPORTED_MIN_VERSION)
+def is_available_gnupg(gnupg):
+  gpg_version_cmd = gnupg + " --version"
+  try:
+    process.run(gpg_version_cmd, stdout=process.PIPE, stderr=process.PIPE)
+    return True
+  except OSError:
+    return False
 
-try:
-  proc = process.run(GPG_VERSION_COMMAND, stdout=process.PIPE,
-    stderr=process.PIPE)
 
-except OSError: # pragma: no cover
-  GPG_COMMAND = "gpg"
-  GPG_VERSION_COMMAND = GPG_COMMAND + " --version"
+GPG_COMMAND = ""
+HAVE_GPG = False
 
-  try:
-    proc = process.run(GPG_VERSION_COMMAND, stdout=process.PIPE,
-      stderr=process.PIPE)
+GPG_ENV_COMMAND = os.environ.get('GNUPG')
+GPG2_COMMAND = "gpg2"
+GPG1_COMMAND = "gpg"
 
-  except OSError:
-    HAVE_GPG = False
+# By default, we allow providing GPG client through the environment
+# assuming gpg2 as default value and test if exists. Otherwise, we assume gpg
+# exists.
+if GPG_ENV_COMMAND:
+  if is_available_gnupg(GPG_ENV_COMMAND):
+    GPG_COMMAND = GPG_ENV_COMMAND
+elif is_available_gnupg(GPG2_COMMAND):
+  GPG_COMMAND = GPG2_COMMAND
+elif is_available_gnupg(GPG1_COMMAND):
+  GPG_COMMAND = GPG1_COMMAND
+
+if GPG_COMMAND:
+  HAVE_GPG = True
```

What is this variable used for?

---

To skip tests and to fail early and gracefully in case someone calls the interface but a gpg command is not available.

I did consider suggesting its removal, now that we can use `if GPG_COMMAND` instead. But I wasn't sure if it's worth the effort. I'm fine either way.

---

That's fine, but it needs a little documentation at the very least, otherwise someone reading the code will repeat the Debian OpenSSL PRNG mistake.